flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -79,6 +83,11 @@ public StreamingMonitorFunction(TableLoader tableLoader, ScanContext scanContext
     this.scanContext = scanContext;
   }
 
+  @Override
+  public void open(Configuration parameters) throws Exception {
```

Comment:
Should the pool size be configured by parameters?

Reply:
Also, is there a way to share pools if there are multiple Iceberg operators in the same Flink job?

Reply:
> Should the pool size be configured by parameters?

Configured from scan context;

> Also, is there a way to share pools if there are multiple Iceberg operators in the same Flink job?

I think it's hard to share, and it will easily get meaningless across distributed nodes.
What do you think, @rdblue?

Reply:
@rdblue do you think that sharing a pool in one job is a blocking issue? If that, we can provide a pool keyed by job. it's somehow reasonable to replace the original pool equivalently  :)

Reply:
Hey, sorry about this. I think my comment here is probably what caused the confusion about sharing pools by job ID. I think there are use cases around this (Steven has one at least) but let's focus on fixing the problem here and sharing resources later.

Thanks for your patience, @yittg!